source/slang/slang-parameter-binding.cpp

```diff
@@ -1072,14 +1072,29 @@ static void addExplicitParameterBindings_GLSL(
     semanticInfo.index = 0;
     semanticInfo.space = 0;
 
+    LayoutSemanticInfo subpassSemanticInfo;
+    bool foundSubpass = false;
+
     if( (foundResInfo = typeLayout->FindResourceInfo(LayoutResourceKind::DescriptorTableSlot)) != nullptr )
     {
-        // Try to find `binding` and `set`
-        if (auto attr = varDecl.getDecl()->findModifier<GLSLBindingAttribute>())
+        for (auto dec : varDecl.getDecl()->modifiers)
         {
-            resInfo = foundResInfo;
-            semanticInfo.index = attr->binding;
-            semanticInfo.space = attr->set;
+            // Try to find `binding` and `set`
+            if (auto glslBindingAttr = as<GLSLBindingAttribute>(dec))
+            {
+                resInfo = foundResInfo;
+                semanticInfo.index = glslBindingAttr->binding;
+                semanticInfo.space = glslBindingAttr->set;
+            }
+            // Try to find `input_attachment_index`
+            if (auto glslAttachmentIndexAttr = as<GLSLInputAttachmentIndexLayoutModifier>(dec))
+            {
+                // Subpass fills semantic info of a descriptor & subpass
+                subpassSemanticInfo.index = stringToInt(glslAttachmentIndexAttr->valToken.getContent());
```

Comment:
We already converted this to int when we check the modifier, right?

Reply:
it is stored as a int literal token (all GLSLLayoutModifier's are)

I wonder if this should be changed though, its significantly slower to call stringToInt 3+ times in a program than it is to cast a int type to another int